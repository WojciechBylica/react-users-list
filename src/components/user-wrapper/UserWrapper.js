import './style.scss';
import PropTypes from 'prop-types';

const UserWrapper = ({ children, link }) => (
  <div className={link ? 'userWrapper userWrapper--link' : 'userWrapper'}>
    {children}
  </div>
);

export default UserWrapper;

UserWrapper.propTypes = {
  children: PropTypes.node,
  link: PropTypes.bool,
};
