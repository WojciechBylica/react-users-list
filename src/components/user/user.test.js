import React from 'react';
import { shallow } from 'enzyme';
import User from '.';
import renderer from 'react-test-renderer';
import UserWrapper from '../user-wrapper';

const user = {
  name: {
    first: 'Dominic',
    last: 'Onderwater',
  },
  location: {
    street: {
      number: 7174,
      name: 'Bontekamp',
    },
    city: 'Pieterburen',
    coordinates: {
      latitude: '44.5991',
      longitude: '-37.1930',
    },
  },
  email: 'dominic.onderwater@example.com',
  login: {
    uuid: '387b2ae0-2637-4599-b725-abb168c1241f',
  },
  registered: {
    date: '2014-12-11T04:02:22.729Z',
  },
  picture: {
    thumbnail: 'https://randomuser.me/api/portraits/thumb/men/6.jpg',
  },
};

//jest snapshot
it('renders ok', () => {
  const tree = renderer.create(<User user={user} />);
  expect(tree).toMatchSnapshot();
});

//enzyme
it('should render h1 with name, image and list with data', () => {
  const user = shallow(<User />);
  expect(user.find(UserWrapper).length).toBe(1);
  expect(user.find('h2').length).toBe(1);
  expect(user.find('ul').length).toBe(1);
});

it('should render h1 with name, image and list with data', () => {
  const component = shallow(<User user={user} />);
  expect(component.find(UserWrapper).length).toBe(1);
  expect(component.find('h2').length).toBe(1);
  expect(component.find('ul').length).toBe(1);
});
