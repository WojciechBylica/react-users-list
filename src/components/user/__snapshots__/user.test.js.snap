// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div
  className="userWrapper"
>
  <h2
    className="user__header"
  >
    <img
      alt="avatar"
      className="user__avatar"
      height={48}
      src="https://randomuser.me/api/portraits/thumb/men/6.jpg"
      width={48}
    />
    <span
      className="user__person"
    >
      Dominic Onderwater
    </span>
  </h2>
  <ul
    className="user__list"
  >
    <li>
      Dominic Onderwater
    </li>
    <li>
      Bontekamp 7174 Pieterburen
    </li>
    <li>
      dominic.onderwater@example.com
    </li>
    <li>
      registered: 
      Thursday, December 11, 2014
    </li>
  </ul>
</div>
`;
